compiler/AST/symbol.cpp

```diff
@@ -468,7 +468,14 @@ const char* Symbol::getSanitizedMsg(std::string msg) const {
 
 void Symbol::maybeGenerateDeprecationWarning(Expr* context) {
   if (!this->hasFlag(FLAG_DEPRECATED)) return;
-
+  // TODO: Remove this exclusion once c_string is deprecated fully.
+  // special handling to stop emitting deprecation warnings about c_string
+  // from within the module that supports c_string.
+  if (context->getModule()->modTag == MOD_INTERNAL &&
+      strcmp(context->getModule()->name, "CString") == 0 &&
+      strcmp(this->name, "c_string") == 0) {
+        return;
+      }
```

Comment:
But there are other internal modules that use `c_string`. Anyway, I thought we were ignoring uses of deprecated symbols in internal modules, but I guess not? Seems like we should use `chpl_c_string` everywhere that the module code needs it?

Reply:
I was worried about the docs exposing the name `chpl_c_string`, but they look correct with the changes so I was able to remove this.